src/validation_workflow/docker/validation_docker.py

```diff
@@ -87,43 +130,50 @@ def validation(self) -> bool:
                     logging.info(f'All tests Pass : {_counter}')
                     return True
                 else:
+                    logging.info(f'Not all tests Pass : {_counter}')
+                    self.cleanup()
                     raise Exception(f'Not all tests Pass : {_counter}')
             else:
                 raise Exception('The container failed to start. Exiting the validation.')
         else:
-            logging.info('\n\nImage digest is Not validated. Exiting..\n\n')
-            raise Exception('Image digest does not match between the opensearchstaging at dockerHub/ecr and opensearchproject at local download. Exiting the validation.')
+            return True
 
     def check_http_request(self) -> bool:
         try:
-            status_code, response_text = ApiTest('https://localhost:9200/').api_get()
-            if status_code == 200:
-                return True
-            else:
-                return False
-        except (requests.exceptions.ConnectionError, requests.exceptions.ConnectTimeout) as e:
-            logging.error(f'Error connecting to https://localhost:9200/: {e}')
+            subprocess.check_output(['curl', 'https://localhost:9200', '-u', 'admin:admin', '--insecure', '-s'])
+            logging.info('status code : 200')
+            return True
+        except subprocess.CalledProcessError as e:
+            logging.error(f'Error connecting to https://localhost:9200: {e}')
             return False
 
     def cleanup(self) -> bool:
-        # clean up
-        if self.cleanup_process():
-            logging.info('cleanup is completed')
-            return True
-        else:
-            logging.info('cleanup is Not completed')
+        try:
+            # clean up
+            if self.args.validate_digest_only:
+                return True
+            if self.cleanup_process():
+                logging.info('cleanup is completed')
+                return True
+            else:
+                logging.info('cleanup is Not completed')
+                return False
+        except Exception as e:
+            logging.error(f'An error occurred during cleanup: {e}')
             return False
+        finally:
+            pass
 
     def cleanup_process(self) -> bool:
         # stop the containers
-        docker_command = f'docker-compose -f {self._target_yml_file} down'
-        result = subprocess.run(docker_command, shell=True, stdout=PIPE, stderr=PIPE, universal_newlines=True)
+        self.docker_compose_down = f'docker-compose -f {self._target_yml_file} down'
+        result = subprocess.run(self.docker_compose_down, shell=True, stdout=PIPE, stderr=PIPE, universal_newlines=True)
 
         # remove docker-compose.yml from the tmp folder
         try:
             os.remove(self._target_yml_file)
         except OSError as e:
-            print("Error: %s - %s." % (e.filename, e.strerror))
+            logging.info("Error: %s - %s." % (e.filename, e.strerror))
```

Comment:
```suggestion
            logging.error("Error: %s - %s." % (e.filename, e.strerror))
```
Also raise an exception that just logging.

Reply:
got it !! makes sense.